docs/guides/nrfconnect_factory_data_configuration.md

```diff
@@ -0,0 +1,297 @@
+
+```
+$ west build -b nrf52840dk_nrf52840 -- --DCONFIG_CHIP_FACTORY_DATA_BUILD=y --DCONFIG_CHIP_DEVICE_DISCRIMINATOR=0xF11
+```
+or by adding them to a ***prj.conf*** file.
+
+### Setting factory data parameters using interactive Kconfig interfaces
+
+All configs can be edited also using interactive Kconfig interface.
+
+See [Configuring nRF Connect examples](../guides/nrfconnect_examples_configuration.md) instruction to get know how to configure Kconfigs.
+
+In configuration window expand an items named `Modules -> connectedhomeip (/home/arbl/matter/connectedhomeip/config/nrfconnect/chip-module) -> Connected Home over IP protocol stack`. After that there should be visible all factory data configs.
+
+```
+(65521) Device vendor ID
+(32774) Device product ID
+[*] Enable Factory Data build
+[*]     Enable merging generated factory data with the build tar
+[*]     Use default certificates located in Matter repository
+[ ]     Enable spake2 verifier generation
+[*]     Enable generation of a new Rotating device id unique id
+(11223344556677889900) Serial number of device
+(Nordic Semiconductor ASA) Human-readable vendor name
+(not-specified) Human-readable product name
+(2022-01-01) Manufacturing date in ISO 8601
+(0) Integer representation of hardware version
+(prerelease) user-friendly string representation of hardware ver
+(0xF00) Device pairing discriminator
+(20202021) Spake2+ passcode
+(1000) Spake2+ iteration count
+(U1BBS0UyUCBLZXkgU2FsdA==) Spake2+ salt in string format
+(uWFwqugDNGiEck/po7KHwwMwwqZgN10XuyBajPGuyzUEV/iree4lOrao5GuwnlQ
+(91a9c12a7c80700a31ddcfa7fce63e44) A rotating device id unique i
+```
+
+> Note: To get more information about how to use Interactive Kconfig interfaces go to [Kconfig docummentation](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/build/kconfig/menuconfig.html).
+
+<hr>
+<a name="Programming factory data"></a>
+
+## Programming factory data
+
+Ready HEX file containing factory data can be programmed into device's FLASH memory using *nrfjprog* and J-Link programmer. To do that use following command:
+
+```
+$ nrfjprog --program factory_data.hex
+```
+
+In this command a *--family* argument can be added and then provide name of board: **NRF52** for nrf52840DK or **NRF53** for nrf5340DK. E.g. 
+
+```
+$ nrfjprog --family NRF52 --program factory_data.hex
+```
+
+> Note: Go [here](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fug_nrf_cltools%2FUG%2Fcltools%2Fnrf_nrfjprogexe.html) to get more information about how to use nrfjprog utility.
+
+
+Another way to program factory data to a device is to use the nrfconnect build system described in [Building an example with factory data](#building-an-example-with-factory-data), and build an example with option `--DCONFIG_CHIP_MERGE_FACTORY_DATA_WITH_FIRMWARE=y`. After that, use the following command from the example's directory to write firmware and newly generated factory data at the same time:
```

Comment:
Just a thought again :D

We have already:
`CONFIG_CHIP_FACTORY_DATA_BUILD`
Could we have the same prefix, or you see it as completely different feature? E.g.:
`CONFIG_CHIP_FACTORY_DATA_MERGE_WITH_FIRMWARE`

Reply:
CONFIG_CHIP_FACTORY_DATA_BUILD config enables generating the factory data during building, but I guess, not everyone will want to merge it in the final merge.hex file. So actually, I added CONFIG_CHIP_FACTORY_DATA_MERGE_WITH_FIRMWARE config as an additional feature for FACTORY_DATA_BUILD.

Reply:
Oh.. what i wanted to ask is if it makes sense to have the same "prefix".

So: `CONFIG_CHIP_MERGE_FACTORY_DATA_WITH_FIRMWARE` =>`CONFIG_CHIP_FACTORY_DATA_MERGE_WITH_FIRMWARE`